test/extended/storage/driver_configuration.go

```diff
@@ -0,0 +1,374 @@
+	}
+
+	e2e.Logf("Validation of %s section in cloud.conf succeeded", sectionName)
+
+	return nil
+}
+
+func validateSnapshotCreation(ctx context.Context, oc *exutil.CLI, successfulSnapshotsCreated int) {
+	e2e.Logf("Validating snapshot creation.")
+
+	pvc, err := createTestPVC(ctx, oc, oc.Namespace(), "test-pvc", "1Gi")
+	o.Expect(err).NotTo(o.HaveOccurred())
+	defer func() {
+		oc.AdminKubeClient().CoreV1().PersistentVolumeClaims(oc.Namespace()).Delete(ctx, pvc.Name, metav1.DeleteOptions{})
+	}()
+
+	pod, err := createTestPod(ctx, oc, pvc.Name, oc.Namespace())
+	o.Expect(err).NotTo(o.HaveOccurred())
+	defer func() { oc.KubeClient().CoreV1().Pods(oc.Namespace()).Delete(ctx, pod.Name, metav1.DeleteOptions{}) }()
+
+	// Wait for pvc to be bound.
+	o.Eventually(func() v1.PersistentVolumeClaimPhase {
+		pvc, err := oc.AdminKubeClient().CoreV1().PersistentVolumeClaims(oc.Namespace()).Get(ctx, "test-pvc", metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		return pvc.Status.Phase
+	}).Should(o.Equal(v1.ClaimBound))
+
+	var wg sync.WaitGroup
+	snapshotsCreated := make(chan string, successfulSnapshotsCreated)
+
+	for i := 0; i < successfulSnapshotsCreated; i++ {
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			snapshotName := fmt.Sprintf("test-snapshot-%d", i)
+			err := createSnapshot(oc, oc.Namespace(), snapshotName, "test-pvc")
+			if err != nil {
+				e2e.Failf("failed to create snapshot: %v", err)
+				return
+			}
+			snapshotsCreated <- snapshotName
+		}(i)
+	}
+	wg.Wait()
+	close(snapshotsCreated)
+
+	defer func() {
+		for snapshotName := range snapshotsCreated {
+			oc.AsAdmin().Run("delete").Args("volumesnapshot", snapshotName).Execute()
+		}
+	}()
+
+	// Wait for snapshots to be readyToUse.
+	o.Eventually(func() (snapshotsReady int) {
+		for snapshotName := range snapshotsCreated {
+			if ready, err := isSnapshotReady(oc, snapshotName); ready {
+				o.Expect(err).NotTo(o.HaveOccurred())
+				e2e.Logf("Snapshot %s is ready", snapshotName)
+				snapshotsReady++
+			} else {
+				e2e.Logf("Snapshot %s is not ready yet", snapshotName)
+			}
+		}
+		return
+
+	}).Should(o.Equal(successfulSnapshotsCreated))
+
+	// Next snapshot creation should be over the set limit and fail.
+	failedSnapshotName := "test-snapshot-failed"
+	err = createSnapshot(oc, oc.Namespace(), failedSnapshotName, "test-pvc")
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	e2e.Logf("Validating that snapshot creation now fails since the limit is reached.")
+	o.Consistently(func() bool {
+		ready, err := isSnapshotReady(oc, failedSnapshotName)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		errMsg, err := oc.Run("get").Args(fmt.Sprintf("volumesnapshot/%s", failedSnapshotName), "-o", "jsonpath={.status.error.message}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		e2e.Logf("Snapshot %s readyToUse: %t, error message: %s", failedSnapshotName, ready, errMsg)
+
+		return strings.Contains(errMsg, "failed to take snapshot of the volume") && !ready
```

Comment:
```suggestion
		return strings.Contains(errMsg, "failed to take snapshot of the volume") && strings.Contains(errMsg, "reaches the configured maximum") && !ready
```
Maybe better to check the failed reason is `reaches the configured maximum`.

Reply:
Changed.